org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/CompilationUnitScope.java

```diff
@@ -765,7 +765,7 @@ ImportBinding[] getDefaultImports() {
 		problemReporter().isClassPathCorrect(
 				TypeConstants.JAVA_LANG_OBJECT,
 			this.referenceContext,
-			this.environment.missingClassFileLocation, false, null/*resolving j.l.O is not specific to any referencing type*/);
+			this.environment, false, null/*resolving j.l.O is not specific to any referencing type*/);
```

Comment:
this change looks unmotivated to me. Wouldn't just changing the signature of `isClassPathCorrect()` simplify things already?
I don't think that method has use for the `LookupEnvironment` itself.